src/main/flight/pid.c

```diff
@@ -538,6 +567,79 @@ static void handleItermRotation()
     }
 }
 
+#ifdef USE_ACRO_TRAINER
+
+int8_t acroTrainerSign(float x)
+{
+    return (x > 0.0f) - (x < 0.0f);
```

Comment:
I don't really like this approach. I would prefer reinterpreting `float` as `int32_t` and then using `i >> 31 ? -1 : 1`
What do you think?

Reply:
I don't think that would exactly work for my case. I need the 0 value result when input is 0.  Special case for when setpoint is not opposing the angle because stick is centered.

I guess it could always be:
```
int acroTrainerSign(int32_t i)
    return i == 0 ? 0 : (i >> 31 ? -1 : 1);
}
```
But I'm not sure that's more efficient.

Reply:
Hmm, also this isn't going to behave well in the region `-1 > x < 1` because the type conversion from `float` to `int32_t`.

Reply:
That's why I said "reinterpet". Reinterpet is either done via a `union` or a `reinterpet_cast`

Reply:
@ledvinap I didn't put it into math.c at this point because I wasn't sure the 3-state (-1, 0, +1) would be generically needed.  I even originally had the macro form:
```#define SIGN(x) ((x > 0.0f) - (x < 0.0f))```
But since it in-lined it lead to code bloat.

In my case the 0 state is needed.  Take the example where currentAngle is positive.  I only want to deactivate the overflow state if the stick is moving the opposite way past zero into negative values.  If the stick is at center (0) then I don't want to deactivate.  `x > 0 ? 1 : -1;` would treat 0 as -1 and would cause deactivation.

Reply:
IMO there little difference between centered stick and stick 1/1000 to opposite side.
With deadband this changes a bit, but it is still border case.

It would be reasonable is there is explicit band around zero that  must be crossed to unlatch.

Reply:
BTW:
`#define SIGN(x) (((x) > 0.0f) - ((x) < 0.0f))` 
consider `SIGN(test ? a : b);`

Reply:
Not sure about the code bloat, I'm quite sure this function will be inlined at most call sites anyway. A function is always preferred over a macro